pycalphad/io/grammar.py

```diff
@@ -6,12 +6,12 @@
 
 pos_neg_int_number = Word('+-' + nums).setParseAction(lambda t: [int(t[0])])  # '+3' or '-2' are examples
 # matching float w/ regex is ugly but is recommended by pyparsing
-regex_after_decimal = r'([0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?)'
+regex_after_decimal = r'([-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?)'
 float_number = Regex(r'[-+]?([0-9]+\.(?!([0-9]|[eE])))|{0}'.format(regex_after_decimal)) \
     .setParseAction(lambda t: [float(t[0])])
 
 chemical_formula = Group(OneOrMore(Word(alphas, min=1, max=2) + Optional(float_number, default=1.0))) + \
-                   Optional(Suppress('/') + pos_neg_int_number, default=0)
+                   Optional(Suppress('/') + float_number, default=0)
```

Comment:
Does your out-of-suite set of TDBs still parse the same with this grammar change?

Reply:
There's no TDB support for any custom TDB syntax in this PR, so this can probably be safely reverted. 

I guess the question is whether it's valid to have floating point charges. My understanding is that the charges are treated like pure elements of `/-` in TDB `SPECIES`, so are floating point charges just as valid as floating point amounts of other pure elements? 

Either way, it's out of scope for this PR, so I'll revert.